R/batch_container.R

```diff
@@ -328,23 +328,63 @@ BatchContainer <- R6::R6Class("BatchContainer",
 
     #' @description
     #' Score current sample assignment,
-    #' @return Returns a vector of all scoring functions values.
-    score = function() {
-      assertthat::assert_that(!is.null(private$scoring_funcs),
-        msg = "Scoring function needs to be assigned"
+    #' @param scoring a function or a names list of scoring functions.
+    #' Each function should return a numeric vector.
+    #' @return Returns a named vector of all scoring functions values.
+    score = function(scoring) {
+      assertthat::assert_that(
+        !missing(scoring),
+        !is.null(scoring),
+        msg = "Scoring function needs to be provided"
       )
-      assertthat::assert_that(is.list(private$scoring_funcs),
-        length(private$scoring_funcs) >= 1,
+      if (is.function(scoring)) {
+        scoring <- list(scoring)
+      } else {
+        assertthat::assert_that(is.list(scoring), length(scoring) >= 1)
+        assertthat::assert_that(
+          all(purrr::map_lgl(scoring, is.function)),
+          msg = "All elements of scoring should be functions"
+        )
+      }
+      if (is.null(names(scoring))) {
+        names(scoring) <- stringr::str_c("score_", seq_along(scoring))
+      }
```

Comment:
Check if all are named?

Reply:
could you please clarify? do you mean that in the case when `names(scoring)` is not `NULL`, I should check names?